### PR TITLE
Use linkTarget in ScaffolderField to enable parameter descriptions to open in a new tab as intended

### LINKED ISSUE #24753: 🐛 Bug Report: Links in mardown description content do not open in new tab
### 📜 Description

When [links](https://github.github.com/gfm/#links) are included in a template yaml definition these are not opened in a new tab.  There is precedence for this being the intent of the application in [github issue #9146](https://github.com/backstage/backstage/issues/9146).  The related PR #9148 provided a linkTarget property that can be passed in.

### 👍 Expected behavior

A github flavored markdown link provided in the description of a scaffolder field should open in a new tab.

### 👎 Actual Behavior with Screenshots

The link is actually opened in the current tab.

image

image

### 👟 Reproduction steps

1. Create a local-template file local-template.yaml in the [plugins/scaffolder-backend/sample-templates folder](https://github.com/backstage/backstage/tree/master/plugins/scaffolder-backend/sample-templates) e.g. 
```
apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: test-form
  title: Test Form
  description: This is a test form
spec:
  owner: guest@example.com
  type: test
  parameters:
    - title: Provide some simple information
      properties:
        prop:
          title: A property with links
          type: string
          description: 'A github flavored markdown formatted [link](https://www.google.com) in the description'
  steps:
    - id: debug
      name: debug
      action: debug:log
      input:
        message: message
  output:
    links:
    - title: Google
      url: https://www.google.com/

```
2. Add `    - ./local-template.yaml` to as defined in [plugins/scaffolder-backend/sample-templates/all-templates.yaml](https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/all-templates.yaml)
3. Run the application locally
4. Route or navigate to http://localhost:3000/create
5. Choose the created form e.g. http://localhost:3000/create/templates/default/test-form
6. Select the github formatted link
7. Observe the link is opened in the current tab

### 📃 Provide the context for the Bug.

When a user clicks a link in the markdown of the description of a template field the link is opened in the current tab.  This causes users to lose progress on current template submissions which is especially frustrating for multi-step templates.

### 🖥️ Your Environment

OS:   Windows_NT 10.0.19045 - win32/x64
node: v18.20.0
yarn: 3.8.1
cli:  0.26.5-next.1 (local)
backstage:  N/A

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by erik-adsk**
## Hey, I just made a Pull Request!

- software-templates: Use linkTarget property of `MarkdownContent` in order to pass "_blank" target to open links (if present) in a new tab

Fixes #24753

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/api-report.md [hunk 1]
```diff
@@ -302,7 +302,10 @@ export const createPublishGitlabMergeRequestAction: (options: {
 }) => TemplateAction_2<
   {
     repoUrl: string;
-    title: string;
+    title: string
+    /**
+     * @public @deprecated use import from \@backstage/plugin-scaffolder-backend-module-github instead
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/api-report.md:307]: Not sure that this is meant to be here?

**Review by benjdlambert — COMMENTED** [review]
Thanks for this, just a comment about some content in the `api-report` which I don't think should be there?

File: plugins/scaffolder-backend/api-report.md (hunk 1 above)
Comment (erik-adsk) [reply plugins/scaffolder-backend/api-report.md]: Yeah, it looks like this was erroneously generated when I used `yarn build:api-reports`.  I've re-generated the api-reports scoping to only the package affected in this PR `yarn build:api-reports plugins/scaffolder-react`.  @benjdlambert please resolve this conversation if you feel this was sufficient.  Thanks!

File: .changeset/loud-pumpkins-bow.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-react': patch
+---
+
+Fixed a bug where links in scaffolder field markdown content e.g. - description was not able to be launched in a new tab. This fix makes it so that all links within Scaffolder Fields are launched in a new tab.
```
Comment (benjdlambert) [inline .changeset/loud-pumpkins-bow.md:5]:
```suggestion
Links that are rendered in the markdown in the `ScaffolderField` component are now opened in new tabs.
```

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks!

**Pull request merged**